webapp/src/ts/modules/tasks/tasks.component.ts

```diff
@@ -0,0 +1,310 @@
+    private translateService: TranslateService,
+  ) {
+    const subscription = combineLatest(
+      this.store.pipe(select(Selectors.getTasksList)),
+      this.store.pipe(select(Selectors.getSelectedTasks)),
+      this.store.pipe(select(Selectors.listContainsTask)),
+      this.store.pipe(select(Selectors.getForms)),
+    ).subscribe(([
+      tasksList,
+      selectedTasks,
+      listContainsTask,
+      forms,
+    ]) => {
+      this.tasksList = tasksList;
+      this.selectedTasks = selectedTasks;
+      this.listContainsTask = listContainsTask;
+      this.forms = forms;
+    });
+    this.subscription.add(subscription);
+
+    this.globalActions = new GlobalActions(store);
+    this.tasksActions = new TasksActions(store);
+    this.servicesActions = new ServicesActions(store);
+  }
+
+  ngOnInit() {
+    this.search();
+
+    const subscription = this.changesService.subscribe({
+      key: 'tasks-list',
+      callback: (change) => {
+        if (change.deleted) {
+          this.tasksActions.removeTaskFromList({ _id: change.id });
+          this.hasTasks = this.tasksList.length;
+          // setActionBarData(); todo
+        } else {
+          this.query({ silent: true, limit: this.tasksList.length });
+        }
+      },
+      filter: (change) => {
+        return change.doc && change.doc.form || this.listContainsTask(change.id);
+      },
+    });
+    this.subscription.add(subscription);
+  }
+
+  ngOnDestroy() {
+    this.subscription.unsubscribe();
+  }
+
+  private prepareTasks(tasks) {
+    return tasks.map(task => {
+      task.icon = task.emission.icon;
+      task.heading = task.emission.contact.name;
+      task.summary = task.emission.title;
+      task.lineage = task.emission.priorityLabel;
+      task.due_date = task.emission.dueDate;
+      return task;
+    });
+  }
+
+  listTrackBy(index, task) {
+    return task._id + task._rev;
+  }
+
+  private query(opts) {
+    const options = Object.assign({ limit: PAGE_SIZE, hydrateContactNames: true }, opts);
+    if (options.limit < PAGE_SIZE) {
+      options.limit = PAGE_SIZE;
+    }
+
+    if (!options.silent) {
+      this.error = false;
+      this.errorSyntax = false;
+      this.loading = true;
+      if (this.selectedTasks.length && isMobile()) {
+        // ctrl.unsetSelected(); todo
+      }
+
+      if (options.skip) {
+        this.appending = true;
+        options.skip = this.tasksList.length;
+      } else if (!options.silent) {
+        this.tasksActions.resetTasksList();
+      }
+    }
+
+    const self = this;
+    return this.rulesEngineService.fetchTaskDocsForAllContacts()
+    //return this.searchService
+    //  .search('tasks', [], options)
+      .then((updatedTasks) => {
+        // add read status todo
+        updatedTasks = this.prepareTasks(updatedTasks);
+
+        self.tasksActions.updateTasksList(updatedTasks);
+        // set action bar data todo
+
+        self.moreItems = updatedTasks.length >= options.limit;
```

Comment:
I don't think we paginate tasks. The `fetchTaskDocsForAllContacts` doesn't accept parameters so there's no place where the pagination can come from.